nanoFramework.Json/JsonSerializer.cs

```diff
@@ -77,26 +77,38 @@ public static string SerializeObject(object o, bool topObject = true)
 
         private static string SerializeClass(object o, Type type)
         {
+            // Cache the type's class-level attributes only if UseIgnoreAttribute setting is enabled.
+            object[] classAttributes = null;
+            if (Settings.UseIgnoreAttribute)
+            {
+                classAttributes = type.GetCustomAttributes(false);
+            }
+
             Hashtable hashtable = new();
 
             // Iterate through all of the methods, looking for internal GET properties
             MethodInfo[] methods = type.GetMethods();
 
             foreach (MethodInfo method in methods)
```

Comment:
One thing here:
For each property you are creating gettersToIgnore based on classAttributes. Which means doing the same work for each property (foreach loop -> ShouldSerializeMethod -> ShouldIgnorePropertyFromClassAttribute -> creating array). Faster approach should be creating ignored property array before ``foreach (MethodInfo method in methods)``.

BTW @josesimoes does nanoFramework support attributes on properties?

Reply:
@torbacz there is nothing preventing you from decorating properties with attributes. 
I guess that you want to know if you can reach custom attributes for properties. That's a different story: _that_ is not supported.

Reply:
Yes, that was my question 😅 Because then we could just decorate each property. Also I'm not quite sure, if we should implement such solution. It's not resolving any problems. If you have class with properties which you don't want to send, just create derived class and pass it to JSON lib.
Don't get me wrong, I like new features but I'm concern about performance, adding new feature where there is a possibility for workaround is making lib much complicated over time.

Reply:
Let's wait for the benchmark results and then decide.
I do share your concern. Impacting performance is something to avoid at all costs. Unless the trade-off it's relevant.

As a side comment: I keep being surprised on how much interest this library has gathered from the community! And the investments on improving it are also surprising to me. 😄